addons/uom/static/src/js/uom_field_widget.js

```diff
@@ -0,0 +1,147 @@
+odoo.define('uom.UomFieldWidget', function (require) {
+"use strict";
+
+var core = require('web.core');
+var session = require('web.session');
+var basic_fields = require('web.basic_fields');
+var fieldUtils = require('web.field_utils');
+var field_registry = require('web.field_registry');
+var utils = require('web.utils');
+
+
+var _t = core._t;
+
+var FieldFloat = basic_fields.FieldFloat;
+var InputField = basic_fields.InputField
+
+var FieldUom = FieldFloat.extend({
+    supportedFieldTypes: ['float', 'uom'],
+    resetOnAnyFieldChange: true, // Have to listen to uom changes
+
+    events: _.extend({}, InputField.prototype.events, {
```

Comment:
try not to use underscore as much as possible (here, Object.assign should be used)